regression-tests/pure2-type-and-namespace-aliases.cpp2

```diff
@@ -30,3 +30,25 @@ main: () = {
     myfunc2 :== myfunc;
     myfunc2();
 }
+
+myclass5: type == myclass4;
+
+myclass3: @struct type = {
+    v0 :== :std::array = (0);;
+    v1: std::array == v0;
+    v2: myclass3 == :myclass3 = ();;
+    v3: _ == v0;
+    // v4 :== v2; // Not (yet) supported.
+    // v5: myclass4 == :myclass4 = ();; // Not yet supported.
+    // v6: myclass5 == :myclass5 = ();; // Not yet supported.
+    // v7 :== myclass3(); // Not (yet) supported.
```

Comment:
Name lookup as added by #506 (for #550) can be used to support these.
For `myclass5`, #533 adds some code to see through aliases.